source/NVDAObjects/UIA/__init__.py

```diff
@@ -998,3 +1007,32 @@ def event_nameChange(self):
 	def event_stateChange(self):
 		return
 
+class SearchField(UIA):
+	"""An edit field that presents suggestions based on search term.
+	"""
+
+	def event_UIA_controllerFor(self):
+		# Only useful if suggestions appear and disappear.
+		focus = api.getFocusObject()
+		if len(self.controllerFor)>0:

```

Comment:
I think you'd want to check that self is focus before checking len(self.controllerFor)... You already fetched focus but didn't use it.